torch/ao/quantization/experimental/observer.py

```diff
@@ -4,44 +4,135 @@
 """
 
 import torch
+import itertools
 from torch.ao.quantization.observer import ObserverBase
 from typing import Tuple
 
 class NonUniformQuantizationObserverBase(ObserverBase):
-    quant_min = None
-    quant_max = None
+    min_val: torch.tensor([])
+    max_val: torch.tensor([])
+    level_indices: torch.tensor([])
+    b: int
+    k: int
+    n: int
+    alpha: float
+    gamma: float
 
     def __init__(
         self,
-        min_val: torch.Tensor,
-        max_val: torch.Tensor,
-        level_indices: torch.Tensor,
-            b: int,
-            k: int) -> None:
-        super().__init__
+        min_val=None,
+        max_val=None,
+        b=0,
+        k=0,
+            dtype=torch.quint8) -> None:
+        super().__init__(dtype)
+        self.min_val = min_val
+        self.max_val = max_val
+        self.level_indices = torch.tensor([])
+        self.b = b
+        self.k = k
+        self.n = 0
+        self.alpha = 0.0
+        self.gamma = 0.0
 
-    def _calculate_qparams(self) -> Tuple[torch.Tensor, torch.Tensor]:
-        pass
+    r""" Calculates nonuniform quantization parameters given min and max value tensors.
+    Args:
+        min_val: Minimum values per channel
+        max_val: Maximum values per channel
+    Returns:
+        gamma: gamma quantization parameter, defined to ensure that alpha is the maximum of the range
+        quantization_levels: non-uniform quantization levels, calculated according to APoT paper:
+        (https://arxiv.org/pdf/1909.13144.pdf)
+        level_indices: int representation of quantization_levels
+    """
+    def _calculate_qparams(
+        self,
+        min_val: torch.tensor,
+        max_val: torch.tensor,
+            signed: bool) -> Tuple[float, torch.tensor, torch.tensor]:
+        # compute alpha
+        self.alpha = max_val
 
-class APoTObserver(NonUniformQuantizationObserverBase):
-    alpha = 0
-    gamma = 0
-    level_indices = torch.Tensor()
+        # compute n and store as member variable
+        if self.k:
+            self.n = self.b // self.k
+
+        if self.n == 0:
+            return (0.0, torch.tensor([]), torch.tensor([]))
```

Comment:
its weird to me that when k isn't specified, we return an empty tensor, seems like k should just be a mandatory argument

Reply:
@asl3 I feel we should throw an exception if `k == 0`